### PR TITLE
bisq-desktop: Nix package improvements

### PR DESCRIPTION
**Pull request by emmanuelrosa**
###### Motivation for this change

 1. Add the Bisq version number to the desktop menu entry.
 2. Fix the *.desktop categories.
 3. Use the embedded Tor binary instead of executing Tor with a script.
 4. Update Bisq from 1.7.0 to 1.7.2
 5. Add Java VM optimizations that can reduce memory consumption starting with Bisq 1.7.2
 6.  Add updater script with signature verification.

###### Minimum version requirement

 Sometimes the Bisq network raises the minimum required Bisq version. Adding the version number so that it's displayed in the desktop environment's menu makes it easy for users to confirm whether they have the necessary version.

###### Free Desktop categories

 Originally, there were two main categories assigned, which violates the Free Desktop standard. This change removes one of the main categories and adds an auxillary category.

###### Return of The Embedded Tor Binary

Bisq comes with an embedded Tor binary (provided by a third party), but we don't use it in this package because it's build for a FHS-abiding Linux distro; Meaning, Tor won't execute because it tried to load libraries from standard locations. To address this problem, the Nix package launches an ephemeral Tor instance for Bisq. The approach works, but it does mean having to manage the tor process, something which is already handled well by Bisq. 

This change modifies the Bisq Jar archive such that it launches the Tor binary from Nixpkgs, allowing Bisq to manage Tor as it does on other Linux distros and operating systems. In a nutshell, when Bisq is launched it extracts a copy of the tor binary from its Jar file and saves it in the Bisq data directory. It is then executed from there. Since Nix doesn't know that Bisq has a runtime dependency on Tor, this change modifies the launcher script to contain a reference to Tor, thus convincing Nix that Tor is a runtime dependency.

###### Updater script

The newly-added updater script allows existing Nix infrastructure to check for Bisq updates and create a corresponding Nixpkgs pulll request. The updater includes signature verification, so that's one less step which needs to be done manually.

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - ~[ ] macOS~
   - [x] other Linux distributions
- ~[ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests]~(https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- ~[ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`~
- [X] Tested execution of all binary files (usually in `./result/bin/`). Successfully completed a trade.
- ~[21.11 Release Notes (or backporting 21.05 Relase notes)]~(https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - ~[ ] (Package updates) Added a release notes entry if the change is major or breaking~
  - ~[ ] (Module updates) Added a release notes entry if the change is significant~
  - ~[ ] (Module addition) Added a release notes entry if adding a new NixOS module~
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 130511` at b5d98048 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>bisq-desktop</li>
  </ul>
</details>
<details>
  <summary>4 suggestions:</summary>
  <ul>
    <li>warning: unused-argument

Unused argument: `psmisc`.
Near pkgs/applications/blockchains/bisq-desktop/default.nix:14:3:
```
   |
14 | , psmisc
   |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Near pkgs/applications/blockchains/bisq-desktop/default.nix:65:3:
```
   |
65 |   installPhase = ''
   |   ^
```
</li>
    <li>warning: unused-argument

Unused argument: `coreutils`.
Near pkgs/applications/blockchains/bisq-desktop/default.nix:11:3:
```
   |
11 | , coreutils
   |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`buildPhase` should probably contain `runHook preBuild` and `runHook postBuild`.

Near pkgs/applications/blockchains/bisq-desktop/default.nix:55:3:
```
   |
55 |   buildPhase = ''
   |   ^
```
</li>
  </ul>
</details>

File: pkgs/applications/blockchains/bisq-desktop/default.nix [hunk 1]
```diff
@@ -60,16 +40,26 @@ stdenv.mkDerivation rec {
...
   unpackPhase = ''
     dpkg -x $src .
   '';
 
+  buildPhase = ''
+    # Replace the embedded Tor binary (which is in a Tar achive)
+    # with one from Nixpkgs.
+
+    mkdir -p native/linux/x64/
+    cp ${tor}/bin/tor ./
```
Comment (roberth) [inline pkgs/applications/blockchains/bisq-desktop/default.nix:58]: It's better not to rely on the property that `tor` still works when copied to a new location. A more robust solution is to package a tiny script that `exec`s the original nixpkgs tor binary in its normal location.

File: pkgs/applications/blockchains/bisq-desktop/default.nix [hunk 2]
```diff
@@ -8,47 +8,27 @@
...
   bisq-launcher = writeScript "bisq-launcher" ''
     #! ${bash}/bin/bash
 
-    # Setup a temporary Tor instance
-    TMPDIR=$(${coreutils}/bin/mktemp -d)
-    CONTROLPORT=$(${coreutils}/bin/shuf -i 9100-9499 -n 1)
-    SOCKSPORT=$(${coreutils}/bin/shuf -i 9500-9999 -n 1)
-    ${coreutils}/bin/head -c 1024 < /dev/urandom > $TMPDIR/cookie
+    # This is just a comment to convince Nix that Tor is a
+    # runtime dependency; The Tor binary is in a *.jar file.
```
Comment (roberth) [inline pkgs/applications/blockchains/bisq-desktop/default.nix:22]:
```suggestion
    # runtime dependency; The Tor binary is in a *.jar file,
    # whereas Nix only scans for hashes in uncompressed text.
```

File: pkgs/applications/blockchains/bisq-desktop/default.nix [hunk 3]
```diff
@@ -60,16 +47,26 @@ stdenv.mkDerivation rec {
...
   '';
 
+  buildPhase = ''
+    # Replace the embedded Tor binary (which is in a Tar achive)
+    # with one from Nixpkgs.
+
+    mkdir -p native/linux/x64/
+    cp ${bisq-tor} ./tor
+    ${gnutar}/bin/tar -cJf native/linux/x64/tor.tar.xz tor
+    ${zip}/bin/zip -r opt/bisq/lib/app/desktop-${version}-all.jar native
```
Comment (roberth) [inline pkgs/applications/blockchains/bisq-desktop/default.nix:67]:
```suggestion
    tar -cJf native/linux/x64/tor.tar.xz tor
    zip -r opt/bisq/lib/app/desktop-${version}-all.jar native
```

These are already in `nativeBuildInputs`, so don't need to be referenced again.

Direct references like these don't work when cross compiling. That doesn't apply directly here, but it's a good habit.

Comment (roberth) [reply pkgs/applications/blockchains/bisq-desktop/default.nix:67]: Same applies for imagemagick below.

**Review by roberth — COMMENTED** [review]
Just a small suggestion; otherwise it looks good to me.

**Comment by emmanuelrosa** [comment]
I rebased unto `master` and added the Bisq 1.7.2 update.

**Comment by emmanuelrosa** [comment]
Rebased on top of #133332

**Comment by veprbl** [comment]
@GrahamcOfBorg eval

**Review by veprbl — APPROVED** [review]
LGTM

File: pkgs/applications/blockchains/bisq-desktop/default.nix [hunk 4]
```diff
@@ -8,70 +8,67 @@
...
-  nativeBuildInputs = [ makeWrapper copyDesktopItems dpkg ];
+  nativeBuildInputs = [ makeWrapper copyDesktopItems imagemagick dpkg gnutar zip xz ];
 
   desktopItems = [
     (makeDesktopItem {
       name = "Bisq";
       exec = "bisq-desktop";
       icon = "bisq";
-      desktopName = "Bisq";
+      desktopName = "Bisq ${version}";
```
Comment (SuperSandro2000) [inline pkgs/applications/blockchains/bisq-desktop/default.nix:52]:
```suggestion
      desktopName = "Bisq;
```

**Pull request merged**